Support experimental enabling of multiple bucket spaces.

geirst opened the pull request:
@vekterli please review
@toregge FYI

geirst committed:
Support experimental enabling of multiple bucket spaces.

vekterli commented on config-model/src/main/java/com/yahoo/vespa/model/content/GlobalDistributionValidator.java:
```diff
@@ -22,9 +22,11 @@
+        if (!enableMultipleBucketSpaces) {
+            verifyGlobalDocumentsHaveRequiredRedundancy(globallyDistributedDocuments, redundancy);
+        }
         verifySearchableCopiesIsSameAsRedundancy(globallyDistributedDocuments, redundancy);
```
Shouldn't this check also be ignored when multiple bucket spaces are enabled? The global distribution config transform will ensure that number of ready copies == redundancy == number of nodes, but this property will not necessarily hold for the node group(s) specified in the XML.

vekterli reviewed (commented):
Let's discuss searchable copies validation first

geirst committed:
Deactivate all validation of redundancy when multiple bucket spaces is enabled.

The global distribution config transform in the distributor will ensure that
number of ready copies == redundancy == number of nodes.

geirst commented:
@vekterli please review again

vekterli reviewed (approved):
:+1:
